src/lib/y2partitioner/widgets/encrypt_method_options.rb

```diff
@@ -210,36 +213,71 @@ def label_widget
 
     # Internal widget to display the pervasive encryption options
     class PervasiveOptions < LuksOptions
+      def initialize(controller, enable: true)
+        super
+        textdomain "storage"
+        self.handle_all_events = true
+      end
+
+      # Handles the events coming from UI, forcing to refresh the encrypt
+      # options each time the encryption method is changed.
+      #
+      # @macro seeCustomWidget
+      def handle(event)
+        if select_master_key? && event["ID"] == master_key_widget.widget_id
+          full_key_widget.refresh(master_key_widget.value)
+          apqn_widget.refresh(master_key_widget.value) if select_apqns?
+          key_type_widget.refresh(candidate_apqns.first.name)
+        end
+
+        nil
+      end
+
       # @return [Boolean]
       def validate
         validate_secure_key_generation
       end
 
+      # Saves the selected APQNs and key type into the controller
+      def store
+        controller.apqns = selected_apqns
+        controller.secure_key_type = key_type_widget.value
+      end
+
       private
 
       # @see LuksOptions#widgets
       def widgets
         widgets = super
-        widgets << apqn_widget if select_apqns?
+        return widgets if exist_secure_key?
 
+        if select_master_key?
+          widgets << master_key_widget
+          widgets << full_key_widget
+        end
+        widgets << apqn_widget if select_apqns?
+        widgets << key_type_widget
```

Comment:
We had similar dynamic widgets in bootloader. Do you test if it fits into 80x24 terminal in ncurses?

Reply:
Yes. It fits.

Reply:
The animations you can see in the PR description are using 80x24.